compiler/rustc_resolve/src/late.rs

```diff
@@ -1550,6 +1567,13 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
         ret
     }
 
+    #[instrument(level = "debug", skip(self))]
+    fn visit_ty_allow_elided_static(&mut self, ty: &'ast Ty) {
+        self.allow_elided_static = true;
+        self.visit_ty(ty);
+        self.allow_elided_static = false;
+    }
```

Comment:
You do not need this method. `allow_elided_static` should be equivalent to the presence of `LifetimeRibKind::Elided(LifetimeRes::Static)` in the stack.

Reply:
What I am trying to do here is to allow not writing `'static` in cases like
```rust
const A: &[u8] = &[];
static B: &str = "hello";
impl Trait{
    const C: &T = ...;
}
```

but still warn about things like
```
warning: elided lifetime has a name
 --> ./test.rs:6:50
  |
6 |     pub fn get_mut(&'static self, x: &mut u8) -> &mut u8 {
  |                                                  ^ this elided lifetime gets resolved as `'static`
  |
```

Reply:
I guess I should've called it `do_not_warn_about_elided_statics`.

Reply:
Done c726445411bcef39af9b4df5f86b0727d86d8929

Reply:
I really mean that this method should not exist. If you want stack-like behaviour, you need to inspect `self.lifetime_ribs`. And if some information is missing, as a field to `LifetimeRibKind`.

Reply:
Fixed 30f5b78eba5f8d8e82b1f41368db9cc08f5fcada